samples/boards/stm32/power_mgmt/blinky_standby_shutdown_mode/src/main.c

```diff
@@ -0,0 +1,168 @@
+ */
+
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/pm/pm.h>
+
+#if !defined(CONFIG_SOC_SERIES_STM32L4X)
+#error Not implemented for other series
+#endif  /* CONFIG_SOC_SERIES_STM32L4X */
+
+#define STACKSIZE 1024
+#define PRIORITY 7
+#define SLEEP_TIME_MS   3000
+
+#include <stm32_ll_pwr.h>
+#define SW0_NODE	DT_ALIAS(sw0)
+#if !DT_NODE_HAS_STATUS(SW0_NODE, okay)
+#error "Unsupported board: sw0 devicetree alias is not defined"
+#endif
+
+/* Semaphore used to control button pressed value */
+static struct k_sem button_sem;
+
+static int led_is_on;
+
+static const struct gpio_dt_spec button =
+	GPIO_DT_SPEC_GET_OR(SW0_NODE, gpios, {0});
+
+static const struct gpio_dt_spec led =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+
+static struct gpio_callback button_cb_data;
+
+void config_wakeup_features(void)
+{
+	/* Configure wake-up features */
+	/* WKUP2(PC13) only , - active low, pull-up */
+	/* Set pull-ups for standby modes */
+	LL_PWR_EnableGPIOPullUp(LL_PWR_GPIO_C, LL_PWR_GPIO_BIT_13);
+	LL_PWR_IsWakeUpPinPolarityLow(LL_PWR_WAKEUP_PIN2);
+	/* Enable pin pull up configurations and wakeup pins */
+	LL_PWR_EnablePUPDCfg();
+	LL_PWR_EnableWakeUpPin(LL_PWR_WAKEUP_PIN2);
+	/* Clear wakeup flags */
+	LL_PWR_ClearFlag_WU();
+}
+
+void button_pressed(const struct device *dev, struct gpio_callback *cb,
+		    uint32_t pins)
+{
+	k_sem_give(&button_sem);
+}
+
+void thread_Shutdown_Standby_mode(void)
+{
+	int sleep_ms;
+	sleep_ms = 300;
+
+	k_sem_init(&button_sem, 0, 1);
+	while (true) {
+		k_sem_take(&button_sem, K_FOREVER);
+		if (led_is_on == false) {
+			gpio_pin_configure(led.port, led.pin, GPIO_DISCONNECTED);
+			printk("user_button pressed\n");
+			printk("enter into Shutdown Mode\n");
+			printk("release the user button to exit from Shutdown Mode\n\n");
+			config_wakeup_features();
+			pm_state_force(0u, &(struct pm_state_info) {PM_STATE_SOFT_OFF, 0, 0});
+
+			k_msleep(200);
+			while (1) {
+			/* stay in Shutdown mode until wakeup line activated */
+				k_msleep(200);
+			}
+		}
+		if (led_is_on == true) {
+			gpio_pin_configure(led.port, led.pin, GPIO_DISCONNECTED);
+			printk("user_button pressed\n");
+			printk("enter into Standby Mode\n");
+			printk("release the user button to exit from Standby Mode\n\n");
+			config_wakeup_features();
+			pm_state_force(0u, &(struct pm_state_info) {PM_STATE_STANDBY, 0, 0});
+
+			k_msleep(200);
+			while (1) {
+			/* stay in Standby mode until wakeup line activated */
+				k_msleep(200);
+			}
+		}
+	k_msleep(sleep_ms);
+	}
+}
+
+K_THREAD_DEFINE(thread_Shutdown_Standby_mode_id, STACKSIZE, thread_Shutdown_Standby_mode,
+	NULL, NULL, NULL, PRIORITY, 0, 0);
+
+void main(void)
+{
+	int ret;
+	uint32_t cause;
+
+	hwinfo_get_reset_cause(&cause);
+	hwinfo_clear_reset_cause();
+
+	if ((LL_PWR_IsActiveFlag_SB() == true) && (cause == 0))	{
```

Comment:
It would be great to have LL_PWR_IsActiveFlag_SB detection as part of `hwinfo_get_reset_cause`.
Can you check if possible to do this ?
(This would be another PR)

Reply:
Yes I will try to check if it possible to do that